core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeUnit.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.scalar;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slice;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Locale.ENGLISH;
+
+public enum DateTimeUnit
```

Comment:
These are supposed to represent the hypothetical fields of a date/time value. They are not units. Therefore, usages that allow for a plural form need to be considered on a case-by-case basis for each function. 

For instance, for the `date_trunc` function is meant to "truncate the timestamp to the nearest X". The plural form of X is incorrect in that context.

Reply:
I'll apply this logic for date_add, date_diff